Add navigationDestination logic for delegations

DRadmir opened the pull request:
Introduces StakeViewModel.navigationDestination(for:) to determine navigation targets based on delegation state. Updates StakeScene to use this method for NavigationLink values, enabling context-aware navigation for different delegation states.

Close: https://github.com/gemwalletcom/gem-ios/issues/1120

DRadmir committed:
Add navigationDestination logic for delegations

Introduces StakeViewModel.navigationDestination(for:) to determine navigation targets based on delegation state. Updates StakeScene to use this method for NavigationLink values, enabling context-aware navigation for different delegation states.

gemcoder21 commented on Features/Staking/Sources/ViewModels/StakeViewModel.swift:
```diff
@@ -135,6 +135,22 @@ public final class StakeViewModel {
         )
     }
+    
+    func navigationDestination(for delegation: StakeDelegationViewModel) -> any Hashable {
```
you need to consolidate this into consistent flow, there is `https://github.com/gemwalletcom/gem-ios/blob/7147a31059770c61e257b93297fd97aab6779f1f/Features/Staking/Sources/ViewModels/StakeViewModel.swift#L115` that has a different handling. it should be consistent for any action

DRadmir committed:
Refactor staking navigation and view model

Replaces custom navigation actions with NavigationLink value-based navigation in StakeScene. Refactors StakeViewModel to remove callback-based navigation, introducing destination-producing methods instead. Updates StakeNavigationView to match new StakeViewModel initializer.

DRadmir committed:
Merge branch 'main' into 1120-skip-stake-details-scene-for-awaiting-withdrawal

gemcoder21 commented on Features/Staking/Sources/Scenes/StakeScene.swift:
```diff
@@ -52,19 +52,17 @@ public struct StakeScene: View {
             if model.showClaimRewards(delegations: delegations) {
-                NavigationCustomLink(
-                    with: ListItemView(
+                NavigationLink(value: model.makeClaimRewardsDestination(delegations: delegations)) {
```
```
DestinationLink<T> {
    value: T
    destination: any Hashable
}
```


```
func claimRewardsDestinationLink() -> DestinationLink {

}
```

DRadmir committed:
Merge branch 'main' of github.com:gemwalletcom/gem-ios into 1120-skip-stake-details-scene-for-awaiting-withdrawal

DRadmir committed:
Refactor staking navigation destination logic

Moved delegation navigation destination logic from StakeViewModel to StakeDelegationViewModel for better encapsulation. Updated StakeScene and StakeViewModel to use the new methods, simplifying navigation link handling and improving code clarity.
